### PR TITLE
makensis: Add optional support for large strings

### PR DESCRIPTION
**Pull request by rhwood**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Adds optional support to create the ["Large Strings" special build](http://nsis.sourceforge.net/Special_Builds#Large_strings) for the current version of makensis.

### CODE REVIEW

**Review by ilovezfs — CHANGES_REQUESTED** [review]
please do not change whitespace in the patch

**Comment by fxcoudert** [comment]
Would there be any downside to making this the default? `makensis` is not very widely used, so it may make sense to avoid having options there if we can

**Comment by rhwood** [comment]
@ilovezfs I've fixed the patch. Sorry about missing that in the original submission.

@fxcoudert The only reason not to make large-strings the default is that the upstream does not make it the default either. Other than that, I have no objection to making large-strings the default. What is the policy on deviating from upstream?

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

@rhwood we do it if we have users for whom the option is useful, if the option introduces no new dependency and there is no drawback or incompatibility for other users

**Comment by rhwood** [comment]
@fxcoudert Thanks! I'll make it default to large strings, allowing users to specify ```--without-long-strings```, but it might take a week since I'm traveling without regular Internet access.

**Comment by ilovezfs** [comment]
I think we should leave it optional unless we're planning to get upstream to change the default.

File: Formula/makensis.rb [hunk 1]
```diff
@@ -42,7 +45,11 @@ def install
     end
 
     # Don't strip, see https://github.com/Homebrew/homebrew/issues/28718
-    scons "STRIP=0", "ZLIB_W32=#{@zlib_path}", "SKIPUTILS=NSIS Menu", "makensis"
+    if build.with? "large-strings"
+      scons "NSIS_MAX_STRLEN=8192", "STRIP=0", "ZLIB_W32=#{@zlib_path}", "SKIPUTILS=NSIS Menu", "makensis"
```
Comment (ilovezfs) [inline Formula/makensis.rb]: This is quite repetitive. It would be good to streamline this with an `args` array.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
That said, @rhwood please open an issue upstream requesting that the default be changed so that we at least have the hope of not carrying a new option indefinitely.

**Comment by rhwood** [comment]
Upstream issue to not need this option is https://sourceforge.net/p/nsis/feature-requests/542/

**Comment by rhwood** [comment]
@ilovezfs Are the changes you requested appropriately addressed?

**Comment by rhwood** [comment]
I've tested that 5523fef builds and runs as expected and added some comments in 334fa9d. Is this ready to merge?

File: Formula/makensis.rb [hunk 2]
```diff
@@ -12,6 +12,9 @@ class Makensis < Formula
     sha256 "5c116395f0a53447856051885ef9c606e60ab8925cb9ec97a21cf72c2c90097d" => :yosemite
   end
 
+  # From http://nsis.sourceforge.net/Special_Builds#Large_strings
```
Comment (vszakats) [inline Formula/makensis.rb]: Request above is still pending. Could you take a look at it?

**Comment by vszakats** [comment]
In newly added comment please change this URL:
`http://nsis.sourceforge.net/Special_Builds#Large_strings`
to this:
`https://nsis.sourceforge.io/Special_Builds#Large_strings`

**Review by vszakats — APPROVED** [review]

**Comment by vszakats** [comment]
Thank you!

**Comment by rhwood** [comment]
I misread the line number in the build report. Sorry about that.

**Comment by fxcoudert** [comment]
Thanks @rhwood for your contribution to Homebrew!

**Comment by idleberg** [comment]
@rhwood Have you considered letting the user specify the string length?

**Example**

```bash
 # default of 8192 bytes
brew install makensis --with-large-strings

# custom string length
brew install makensis --with-large-strings=16384
```

This might also require a change of the wording (of the option) to communicate this better, I don't know.

**Comment by rhwood** [comment]
I had not; I made makensis installed with the ```--with-large-strings``` argument match the large-strings special build [distributed by NSIS](http://nsis.sourceforge.net/Special_Builds#Large_strings).

**Comment by idleberg** [comment]
I don't want to overcomplicate things. I don't think many people would use this, but I remember seeing questions regarding this on the NSIS forum. Let's see what the maintainers think.

**Comment by fxcoudert** [comment]
Let's keep a single option, matching the NSIS build.

**Pull request merged**